readme.txt
* OVERVIEW

Zekr is an Open Source Qur'an study project. This project is planned to be a
universal, Open Source, and cross-platform application to ease most of
the usual refers to the holy Qur'an. The main idea is to build an as generic
as possible platform to accept different Qur'anic resources.

Copyright (c) 2004-2012 zekr.org, and other contributors.

Website: http://zekr.org.
Translations, recitations and other resources: http://zekr.org/resources.html.

Join Zekr Google group (and mailing list) by sending a message to:
	zekr-subscribe@googlegroups.com


* LICENSE

Original Zekr source code ([zekr]/dist/zekr-src.jar) and other artifacts
written by its author (*.vm, *.js, *.css, *.html, *.xslt, *.xml) are licensed
under GNU GPL v.2, which is available here: [zekr]/doc/zekr-license.txt.

Tanzil Quran text files 1.0.2 (quran-simple.txt and quran-uthmani.txt) are bundled 
under the following agreement: [zekr]/doc/tanzil-license.txt.

CRCIS Quran root and stem list, ver. 0.9.1 (quran-root.txt) is licensed under the following
agreement: [zekr]/doc/quran-root-crcis-license.txt.

Most of images and icons used in this application are taken from Linux icon packs,
which are under Lesser General Public License 2.1, [zekr]/doc/lgpl-2.1.txt.
Others are either unspecified or made by Zekr author under GPLv2.

Zekr splash screen (zekr-splash-alpha.png) is created by Bandar Raffa
(http://bandar.raffah.com), specially for the Zekr project, under
creativecommons by-nc-nd 3.0 (non-free) license.

zekr.exe and zekrc.exe are based on the native Java wrapper library WinRun4j, which is under
the Common Public License (CPL) 1.0, [zekr]/doc/cpl-1.0.txt.

For other libraries and dependencies come with standard Zekr releases see
their corresponding licenses under doc directory:
- Apache commons (including parts of commons-io-1.4, commons-lang-2.4,
    commons-logging-1.0.4, commons-collections-3.2.1, commons-codec-1.3,
    and commons-configuration-1.6):
    Apache License version 2.0, [zekr]/doc/apache-license-2.0.txt
- jintellitype-1.3.8.jar: Apache License version 2.0, [zekr]/doc/apache-license-2.0.txt
- Lucene libraries (lucene-core-3.0.0.jar, lucene-highlighter-3.0.0.jar,
    lucene-snowball-3.0.0.jar, lucene-memory-3.0.0.jar, and lucene-misc-3.0.0.jar):
    Apache License version 2.0, [zekr]/doc/apache-license-2.0.txt
- log4j-1.2.8: Apache Software License version 1.1, [zekr]/doc/apache-software-license-1.1.txt
- velocity-1.6.2: Apache License version 2.0, [zekr]/doc/apache-license-2.0.txt
- swt.jar: Eclipse Public License 1.0, [zekr]/doc/epl-1.0.txt
- JavaZoom libraries (jlayer-1.0.1.jar, mp3spi-1.9.4.jar, basicplayer-3.0.jar, and
    vorbisspi-1.0.3.jar): Lesser General Public License 2.1, [zekr]/doc/lgpl-2.1.txt
- jorbis-0.0.17.jar: Lesser General Public License 2.1, [zekr]/doc/lgpl-2.1.txt
- jspeex-0.9.7.jar: Modified BSD license, [zekr]/doc/jspeex-license.txt
- tritonus-share-0.3.6.jar: General Public License 2.1, [zekr]/doc/gpl-2.0.txt
- tritonus-jorbis-0.3.6.jar: General Public License 2.1, [zekr]/doc/gpl-2.0.txt

The following fonts are bundled in Zekr packages, but are not required:
- Zekr Quran font (ZekrQuran.ttf and ZekrQuranBold.ttf) is under SIL.org Open Font License 1.1,
    [zekr]/doc/sil-ofl-1.1.txt. Copyright (c) 2010-2012, Muhammet Abay (muhabay@gmail.com).
- Scheherazade font (ScheherazadeRegOT.ttf) is under SIL.org Open Font License 1.1,
    [zekr]/doc/sil-ofl-1.1.txt
- me_quran font (me_quran_volt_newmet.ttf) is licensed by Meor Ridzuan and permission
    ganted to be bundled in Zekr.
- KFGQPC Uthman Taha Naskh (UthmanTN1 Ver10.otf and UthmanTN1B Ver10.otf) is built based on
    Uthman Taha calligraphy (Calligrapher of Mushaf Al-Madina) by King Fahd Glorious Quran
    Printing Complex (KFGQPC). This font can be distributed freely for both commercial and
    non-commercial use, according to the Arabic license terms available here:
    http://fonts.qurancomplex.gov.sa/?page_id=40.


doc/changes.txt
Version 0.1.0 beta 1 - [25-3-05]
Version 0.1.0 beta 2 - [29-3-05]

Version 0.1.0 release - [13-9-05] (First stable release)
Features/Enhancements:
	Nullsoft Installer (multilingual) support
	Basic navigation over suras and ayas
	.exe launcher for windows systems
	Sura ayas view layout: line-by-line, block
	Display sura info (descent type, aya count, sura number, etc.)
	Cache visited suras in a local directory for faster retrieval
Bug fixes:
	Problem with run.bat (JAVA_HOME) was fixed.
	Enter (Return) will work in Quran Form. (sura/aya selection)
	Localized hint for juz numbers
	Splash screen has title as well as icons.

Version 0.2.0 b1 - [9-12-05]
Features/Enhancements:
	Print a Quran view
	Search the whole Quran and also current view (a search result or a sura) with matching and non-matching diacritics
	Simple search statistics.
	Help menu: Zekr homepage, About.
	Language switch menu item for changing UI language while application is running through view menu.
	Configuration (layout, language, ...) is now saved in the xml file
Bug fixed:
	All misspelling for the wrong term sujda was changed to sajda.

Version 0.2.0 b2 - [24-12-05]
Features/Enhancements:
	Simple Save as menu for exporting a single sura or a search result
	Five language packs: Farsi (by Zekr team), Arabic (by Younes Moumen), English, French (by Younes Moumen), Indonesian (by Ari R. Fikri).
	Aya navigation enhancement: choosing to show an aya will also displays some previous ayas.
	Country flag icon for each language

Version 0.2.0 release - [20-1-06]
Features/Enhancements:
	Added German language pack (translated by Riyad Preukschas)
	Hyperlink for direct navigating to search result location (sura, aya) in the Quran
	More smoother Bismillah image.
	Enhanced presentation for sajda sign (character \u6E9)
	Text box replace with an editable combo box (can hold up to 40 search history).
	A Linux port is available (Mozilla 1.4+ GTK2 must be installed)
	A Mac port is available. (25-2-06)
Bug fixes:
	Unicode ALEF_MADDA (0x622) is now acted like alef when searching with non-matching diacritics
	All the following misspelling was changed from column one to column two:
	soora		sura
	joz			juz
	Search problem. ==> (أأنذرتهم was found but أنذرتهم could not be found)

Version 0.3.0 b1 - [14-3-06]
Features/Enhancements:
	Log enhancements. More info about system (OS, JRE, locale, ...) are stored.
	About menu enhancement.
	Menu items: Translation and Theme under View menu.
	Translation support:
		Basic support for translation packs.
		2 Different views for translation: 2 frames and inline.
		Ayatollah Makarem Shirazi's translation is added under a non-commercial use license
		A number of enhancements for BIDI support for translations view
	Theme support:
		A basic default theme (sky) is available.
		All themes are read at startup from [zekr]/res/theme, so every folder there is a potential theme!
		All themes should have a "theme.properties" file inside their folder (theme folder name is equal to theme ID).
		Does not have a GUI for setting options.
	InitRuntime changed to ApplicationRuntime, and is loaded once and can be accessed from ApplicationConfig.
	Translation layout (block/line-by-line) can be set as well as Quran layout.
	Tools>Add>Translation and Tools>Add>Themes. two menu items for importing translation and theme pack.
	M.H.Shakir's English translation will be available in the standard Zekr release under a non-commercial license.
	Apache Jakarta commons are stored in a single jar file, inside [zekr]/lib. Extra unused classes are removed from the jar file.
	Last seen Quran location is stored in zekr-config.xml, so that the next time Zekr starts, go to that location.
Bug fixes:
	JDK 1.4.2's JAXP SAX does not support BOM, so all language pack BOMs were removed.

Version 0.3.0 b2 - [25-3-06]
Features/Enhancements:
	Show popup translation for search result.
	4 different layouts:
		1. Quran only layout. show Quran only. Full Quran search is allowed in this view.
		2. Translation only layout. Full Quran search is not allowed, but one can search the current view.
		3. Separate; Quran at the top, translation at the bottom in 2 separate frames. Search is fully supported.
		4. Mixed; Quran and translation are mixed aya by aya. Each aya will be followed by its translation. Search is fully supported.
	Consequently, 4 new menu items under view>layout.
	Simple Options form under tools>options, with 2 different tabs:
		1. General tab: currently one option for enable/disable splash screen.
		2. View tab: a grid of key-value pairs for current theme. CRUD (Create, Read, Update and Delete) is fully supported on the grid.
Bug fixes:
	ALEF_MAKSURA will be treated as ARABIC_YEH when diacritic match is not set. In fact all YEH styled characters 
			(ARABIC_YEH, FARSI_YEH and ALEF_MAKSURA) both in Quran and search box are converted to ARABIC_YEH.
	A fine typo in Shakir translation was fixed.

Version 0.3.0 release - [25-5-06]
Features/Enhancements:
	All paths are now passed as URIs (instead of File.getAbsolutioPath()) to Velocity.
	Flag icons updated to more nice looking ones.
	Added Russian language pack (translated by Michael Kalinin).
	3 Russian translations (Porokhova, Osmanov, and Kuliev) provided by Michael Kalinin (under a non-commercial license). Will be as translation pack in the website (not present in the default distribution).
	The Quran text file completely changed from the old one to a more accurate one (after finding two typos in the old text file)
	View>Language moved to Tools>Options form, and now has a more nice look.
	View>Theme moved to Tools>Options form.
	Fooladvand (Farsi) translation will be available under a non-commercial license.
	A new sample theme (id: sample) is available.
	Max search result (default: 200) can be set on Options form.
	Updated to SWT build 3232.
	Permission to use Kuliev's translation in Zekr (done by Michael Kalinin).
Bug fixes:
	A typo in Makarem translation of sura Noor, aya 26 has been fixed.

res/ui/js/ifx.js
/**
 * Interface Elements for jQuery
 * FX
 * 
 * http://interface.eyecon.ro
 * 
 * Copyright (c) 2006 Stefan Petre
 * Dual licensed under the MIT (MIT-LICENSE.txt) 
 * and GPL (GPL-LICENSE.txt) licenses.
 *   
 * $Revision$
 * $Log$
 * Revision 1.4  2007/05/01 16:29:39  mohsen_s
 * Zekr 0.6.0 beta 1
 *
 * Revision 1.2  2007/01/03 12:14:28  Mohsen Saboorian
 * *** empty log message ***
 *
 * Revision 1.1  2006/09/23 07:51:00  Mohsen Saboorian
 * *** empty log message ***
 *
 * Revision 1.10  2006/09/01 21:23:51  Stef
 * Fixed getting the sizes of the element to wrap
 *
 * Revision 1.9  2006/08/31 16:08:15  Stef
 * Added dual licensing
 *
 * Revision 1.8  2006/08/29 18:46:32  Stef
 * Changed the way fx wrapper is destroyed. first apply old style, then move element
 *
 * Revision 1.7  2006/08/27 19:05:55  Stef
 * *** empty log message ***
 *
 * Revision 1.6  2006/08/26 09:34:47  Stef
 * *** empty log message ***
 *
 * Revision 1.5  2006/08/26 09:31:11  Stef
 * *** empty log message ***
 *
 * Revision 1.4  2006/08/26 09:02:56  Stef
 * Added modified fx function from jQuery 1.0
 *
 * Revision 1.3  2006/08/03 21:38:41  Stef
 * *** empty log message ***
 *
 * Revision 1.2  2006/08/02 17:59:24  Stef
 * *** empty log message ***
 *
 */

//overwrite jQuery's default fx function with the new one to support diferent transitions
jQuery.fx = function( elem, options, prop, transition ){

	var z = this;

	z.transition = /easein|easeout|easeboth|bouncein|bounceout|bounceboth|elasticin|elasticout|elasticboth/.test(transition) ? transition : 'original';
	// The users options
	z.o = {
		duration: options.duration || 400,
		complete: options.complete,
		step: options.step
	};

	// The element
	z.el = elem;

	// The styles
	var y = z.el.style;

	// Simple function for setting a style value
	z.a = function(){
		if ( options.step )
			options.step.apply( elem, [ z.now ] );

		if ( prop == "opacity" ) {
			if (z.now == 1) z.now = 0.9999;
			if (window.ActiveXObject)
				y.filter = "alpha(opacity=" + z.now*100 + ")";
			else
				y.opacity = z.now;

		// My hate for IE will never die
		} else if ( parseInt(z.now) )
			y[prop] = parseInt(z.now) + "px";
			
		y.display = "block";
	};

	// Figure out the maximum number to run to
	z.max = function(){
		return parseFloat( jQuery.css(z.el,prop) );
	};

	// Get the current size
	z.cur = function(){
		var r = parseFloat( jQuery.curCSS(z.el, prop) );
		return r && r > -10000 ? r : z.max();
	};

	// Start an animation from one number to another
	z.custom = function(from,to){
		z.startTime = (new Date()).getTime();
		z.now = from;
		z.a();

		z.timer = setInterval(function(){
			z.step(from, to);
		}, 13);
	};

	// Simple 'show' function
	z.show = function( p ){
		if ( !z.el.orig ) z.el.orig = {};

		// Remember where we started, so that we can go back to it later
		z.el.orig[prop] = this.cur();

		z.custom( 0, z.el.orig[prop] );

		// Stupid IE, look what you made me do
		if ( prop != "opacity" )
			y[prop] = "1px";
	};

	// Simple 'hide' function
	z.hide = function(){
		if ( !z.el.orig ) z.el.orig = {};

		// Remember where we started, so that we can go back to it later
		z.el.orig[prop] = this.cur();

		z.o.hide = true;

		// Begin the animation
		z.custom(z.el.orig[prop], 0);
	};

	// IE has trouble with opacity if it does not have layout
	if ( jQuery.browser.msie && !z.el.currentStyle.hasLayout )
		y.zoom = "1";

	// Remember  the overflow of the element
	if ( !z.el.oldOverlay )
		z.el.oldOverflow = jQuery.css( z.el, "overflow" );

	// Make sure that nothing sneaks out
	y.overflow = "hidden";

	// Each step of an animation
	z.step = function(firstNum, lastNum){
		var t = (new Date()).getTime();

		if (t > z.o.duration + z.startTime) {
			// Stop the timer
			clearInterval(z.timer);
			z.timer = null;

			z.now = lastNum;
			z.a();
			if (z.el.curAnim)
				z.el.curAnim[ prop ] = true;
			
			var done = true;
			for ( var i in z.el.curAnim ) {
				if ( z.el.curAnim[i] !== true )
					done = false;
			}
					
			if ( done ) {
				// Reset the overflow
				y.overflow = z.el.oldOverflow;
			
				// Hide the element if the "hide" operation was done
				if ( z.o.hide ) 
					y.display = 'none';
				
				// Reset the property, if the item has been hidden
				if ( z.o.hide ) {
					for ( var p in z.el.curAnim ) {
						y[ p ] = z.el.orig[p] + ( p == "opacity" ? "" : "px" );

						// set its height and/or width to auto
						if ( p == 'height' || p == 'width' )
							jQuery.setAuto( z.el, p );
					}
				}
			}
			
			// If a callback was provided, execute it
			if( done && z.o.complete && z.o.complete.constructor == Function )
				// Execute the complete function
				z.o.complete.apply( z.el );
		} else {
			var n = t - this.startTime;
			// Figure out where in the animation we are and set the number
			//var p = (t - this.startTime) / z.o.duration;
			var p = n / z.o.duration;
			//z.now = ((-Math.cos(p*Math.PI)/2) + 0.5) * (lastNum-firstNum) + firstNum;
			z.now = jQuery.fx.transitions(p, n, firstNum, (lastNum-firstNum), z.o.duration, z.transition);

			// Perform the next step of the animation
			z.a();
		}
	};	
};

jQuery.fxCheckTag = function(e)
{
	if (/tr|td|tbody|caption|thead|tfoot|col|colgroup|th|body|header|script|frame|frameset|option|optgroup|meta/i.test(e.nodeName) )
		return false;
	else 
		return true;
};
jQuery.fx.destroyWrapper = function(e, old)
{
	c = e.firstChild;
	cs = c.style;
	cs.position = old.position;
	cs.marginTop = old.margins.t;
	cs.marginLeft = old.margins.l;
	cs.marginBottom = old.margins.b;
	cs.marginRight = old.margins.r;
	cs.top = old.top + 'px';
	cs.left = old.left + 'px';
	e.parentNode.insertBefore(c, e);
	e.parentNode.removeChild(e);
};
jQuery.fx.buildWrapper = function(e)
{
	if (!jQuery.fxCheckTag(e))
		return false;
	var t = jQuery(e);
	var es = e.style;
	var restoreStyle = false;
	oldStyle = {};
	oldStyle.position = t.css('position');
	oldStyle.sizes = jQuery.iUtil.getSize(e);
	oldStyle.margins = jQuery.iUtil.getMargins(e);
	if (t.css('display') == 'none') {
		oldVisibility = t.css('visibility');
		t.show();
		restoreStyle = true;
	}
	oldStyle.top = parseInt(t.css('top'))||0;
	oldStyle.left = parseInt(t.css('left'))||0;
	if (restoreStyle) {
		t.hide();
		es.visibility = oldVisibility;
	}
	var wid = 'w_' + parseInt(Math.random() * 10000);
	var wr = document.createElement(/img|br|input|hr|select|textarea|object|iframe|button|form|table|ul|dl|ol/i.test(e.nodeName) ? 'div' : e.nodeName);
	jQuery.attr(wr,'id', wid);
	jQuery(wr).addClass('fxWrapper');
	var wrs = wr.style;
	var top = 0;
	var left = 0;
	if (oldStyle.position == 'relative' || oldStyle.position == 'absolute'){
		top = oldStyle.top;
		left = oldStyle.left;
	}
	
	wrs.top = top + 'px';
	wrs.left = left + 'px';
	wrs.position = oldStyle.position != 'relative' && oldStyle.position != 'absolute' ? 'relative' : oldStyle.position;
	wrs.height = oldStyle.sizes.hb + 'px';
	wrs.width = oldStyle.sizes.wb + 'px';
	wrs.marginTop = oldStyle.margins.t;
	wrs.marginRight = oldStyle.margins.r;
	wrs.marginBottom = oldStyle.margins.b;
	wrs.marginLeft = oldStyle.margins.l;
	wrs.overflow = 'hidden';
	if (jQuery.browser == "msie") {
		es.filter = "alpha(opacity=" + 0.999*100 + ")";
	}
	es.opacity = 0.999;
	//t.wrap(wr);
	e.parentNode.insertBefore(wr, e);
	wr.appendChild(e);
	es.marginTop = '0px';
	es.marginRight = '0px';
	es.marginBottom = '0px';
	es.marginLeft = '0px';
	es.position = 'absolute';
	es.listStyle = 'none';
	es.top = '0px';
	es.left = '0px';
	return {oldStyle:oldStyle, wrapper:jQuery(wr)};
};
jQuery.fx.transitions = function(p, n, firstNum, delta, duration, type)
{
	if (type == 'original') {
		return ((-Math.cos(p*Math.PI)/2) + 0.5) * delta + firstNum;
	}
	if (type == 'easein') {
		return delta*(n/=duration)*n*n + firstNum;
	}
	if (type == 'easeout') {
		return -delta * ((n=n/duration-1)*n*n*n - 1) + firstNum;
	}
	if (type == 'easeboth') {
		if ((n/=duration/2) < 1)
			return delta/2*n*n*n*n + firstNum;
			return -delta/2 * ((n-=2)*n*n*n - 2) + firstNum;
	}
	if (type == 'bounceout') {
		if ((n/=duration) < (1/2.75)) {
			return delta*(7.5625*n*n) + firstNum;
		} else if (n < (2/2.75)) {
			return delta*(7.5625*(n-=(1.5/2.75))*n + .75) + firstNum;
		} else if (n < (2.5/2.75)) {
			return delta*(7.5625*(n-=(2.25/2.75))*n + .9375) + firstNum;
		} else {
			return delta*(7.5625*(n-=(2.625/2.75))*n + .984375) + firstNum;
		}
	}
	if (type == 'bouncein') {
		nm = duration - n;
		if ((nm/=duration) < (1/2.75)) {
			m = delta*(7.5625*nm*nm);
		} else if (nm < (2/2.75)) {
			m = delta*(7.5625*(nm-=(1.5/2.75))*nm + .75);
		} else if (nm < (2.5/2.75)) {
			m = delta*(7.5625*(nm-=(2.25/2.75))*nm + .9375);
		} else {
			m = delta*(7.5625*(nm-=(2.625/2.75))*nm + .984375);
		}
		return delta - m + firstNum;
	}
	
	if (type == 'bounceboth') {
		if (n < duration/2) {
			nm = n * 2;
			if ((nm/=duration) < (1/2.75)) {
				m = delta*(7.5625*nm*nm);
			} else if (nm < (2/2.75)) {
				m = delta*(7.5625*(nm-=(1.5/2.75))*nm + .75);
			} else if (nm < (2.5/2.75)) {
				m = delta*(7.5625*(nm-=(2.25/2.75))*nm + .9375);
			} else {
				m = delta*(7.5625*(nm-=(2.625/2.75))*nm + .984375);
			}
			return (delta - m + firstNum) * .5 + firstNum;
		} else {
			n = n * 2 - duration;
			if ((n/=duration) < (1/2.75)) {
				m = delta*(7.5625*n*n) + firstNum;
			} else if (n < (2/2.75)) {
				m = delta*(7.5625*(n-=(1.5/2.75))*n + .75) + firstNum;
			} else if (n < (2.5/2.75)) {
				m = delta*(7.5625*(n-=(2.25/2.75))*n + .9375) + firstNum;
			} else {
				m = delta*(7.5625*(n-=(2.625/2.75))*n + .984375) + firstNum;
			}
			return m * .5 + delta*.5 + firstNum;
		}
	}
	if (type == 'elasticout') {
		if ((n/=duration)==1)
			return firstNum+delta;
		return delta*Math.pow(2,-10*n) * Math.sin( (n*duration-(duration*.3)/4)*(2*Math.PI)/(duration*.3) ) + delta + firstNum;
	}
	if (type == 'elasticin') {
		if (n==0)
			return b;  
		if ((n/=duration)==1) 
			return firstNum+delta;
		return -(delta*Math.pow(2,10*(n-=1)) * Math.sin( (n*duration-(duration*.3)/4)*(2*Math.PI)/(duration*.3) )) + firstNum;
	}
	if (type == 'elasticboth') {
		if (n==0)
			return firstNum;
		if ((n/=duration)==1) 
			return firstNum+delta;
		$('#test').html(p +'<br />'+n);
		if (p < 1)
			return -.5*(delta*Math.pow(2,10*(n-=1)) * Math.sin( (n*duration-(duration*.45)/4)*(2*Math.PI)/(duration*.45) )) + firstNum;
		return delta*Math.pow(2,-10*(n-=1)) * Math.sin( (n*duration-(duration*.45)/4)*(2*Math.PI)/(duration*.45) )*.5 + delta + firstNum;
	}
			
};